examples/magic_image_arithmetic.py

```diff
@@ -0,0 +1,34 @@
+"""Basic example of using magicgui to create an Image Arithmetic GUI in napari."""
+import enum
+import numpy as np
+import napari
+
+
+# Enums are a convenient way to get a dropdown menu
+class Operation(enum.Enum):
+    """A set of valid arithmetic operations for image_arithmetic."""
+    add = np.add
+    subtract = np.subtract
+    multiply = np.multiply
+    divide = np.divide
+
+
+# Define our function.
+def image_arithmetic(layerA: napari.layers.Image, operation: Operation, layerB: napari.layers.Image) -> napari.layers.Image:
```

Comment:
I'm trying this again after the merge of magicGui PR43, and if I use strings for types here instead of the classes I get the following error (I'm not sure if that is expected or not, so apologies if this is expected)

```python
(base) czirwc1macos2701:napari nsofroniew$ python examples/magic_image_arithmetic.py
ERROR:root:Unhandled exception:
Traceback (most recent call last):
  File "/Users/nsofroniew/GitHub/napari/napari/_qt/event_loop.py", line 79, in gui_qt
    yield app
  File "examples/magic_image_arithmetic.py", line 34, in <module>
    viewer.window.add_function_widget(image_arithmetic, magic_kwargs=magic)
  File "/Users/nsofroniew/GitHub/napari/napari/_qt/qt_main_window.py", line 654, in add_function_widget
    widget = magicgui(**magic_kwargs)(function)
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/function_gui.py", line 288, in inner_func
    app=app,
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/function_gui.py", line 79, in __init__
    widgets=list(sig.widgets(app).values()),
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/signature.py", line 220, in widgets
    {n: p.to_widget(app) for n, p in self.parameters.items()}
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/signature.py", line 220, in <dictcomp>
    {n: p.to_widget(app) for n, p in self.parameters.items()}
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/signature.py", line 144, in to_widget
    options=options,
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/widgets/_bases.py", line 229, in create
    wdg_class, opts = get_widget_class(default, annotation, options)
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/type_map.py", line 156, in get_widget_class
    widget_type, _options = pick_widget_type(value, annotation, _options)
  File "/Users/nsofroniew/GitHub/magicgui/magicgui/type_map.py", line 131, in pick_widget_type
    raise ValueError(f"Could not pick widget for type: {dtype!r}")
ValueError: Could not pick widget for type: ForwardRef('napari.layers.Image')
```

I wonder if that is something that we need to fix on the magicgui side or on the napari side. This was raised initially https://github.com/napari/magicgui/issues/46 which I will acknowledge is still open

Reply:
Still need to add that on the magicgui side, your napari/magicgui#46 issue is still very much valid!  Shouldn't be hard now though, hopefully just a few lines.  Will do it soon

Reply:
added in https://github.com/napari/magicgui/pull/52

Reply:
I'n still getting an error on this, but it looks like it is on the "napari" side with something in `utils._magicgui` file

```python-traceback
(base) czirwc1macos2701:napari nsofroniew$ python examples/magic_parameter_sweep.py
/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/widgets/_bases.py:217: FutureWarning:

FloatSlider.__init__() got unexpected keyword arguments {'maximum'}.
In the future this will raise an exception

  FutureWarning,
/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/widgets/_bases.py:722: FutureWarning:

As of magicgui 0.2.0, a `choices` callable may accept only a single positional
argument (an instance of `magicgui.widgets.CategoricalWidget`), and must return
an iterable (the choices to show). Function 'get_layers' accepts 2 arguments.
In the future, this will raise an exception.

  FutureWarning,
ERROR:root:Unhandled exception:
Traceback (most recent call last):
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/widgets/_bases.py", line 710, in choices
    _choices = choices(self)
  File "/Users/nsofroniew/GitHub/napari/napari/utils/_magicgui.py", line 130, in get_layers
    layer for layer in viewer.layers if isinstance(layer, layer_type)
  File "/Users/nsofroniew/GitHub/napari/napari/utils/_magicgui.py", line 130, in <genexpr>
    layer for layer in viewer.layers if isinstance(layer, layer_type)
TypeError: isinstance() arg 2 must be a type or tuple of types

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/widgets/_bases.py", line 238, in <lambda>
    lambda *x: self.parent_changed(value=self.parent)
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/events.py", line 603, in __call__
    self._invoke_callback(cb, event)
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/events.py", line 625, in _invoke_callback
    cb_event=(cb, event),
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/events.py", line 619, in _invoke_callback
    cb(event)
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/widgets/_bases.py", line 686, in reset_choices
    self.choices = self._default_choices
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/widgets/_bases.py", line 725, in choices
    _choices = choices(self.native, self.annotation)  # type: ignore
  File "/Users/nsofroniew/GitHub/napari/napari/utils/_magicgui.py", line 130, in get_layers
    layer for layer in viewer.layers if isinstance(layer, layer_type)
  File "/Users/nsofroniew/GitHub/napari/napari/utils/_magicgui.py", line 130, in <genexpr>
    layer for layer in viewer.layers if isinstance(layer, layer_type)
TypeError: isinstance() arg 2 must be a type or tuple of types
```